packages/suite/src/views/wallet/transactions/TransactionList/TransactionsGroup/TransactionsGroup.tsx

```diff
@@ -36,9 +45,24 @@ export const TransactionsGroup = ({
     ...rest
 }: TransactionsGroupProps) => {
     const [isHovered, setIsHovered] = useState(false);
+    const historicFiatRates = useSelector(selectHistoricFiatRates);
     const totalAmountPerDay = sumTransactions(transactions);
-    const totalFiatAmountPerDay = sumTransactionsFiat(transactions, localCurrency);
-    const isMissingFiatRates = transactions.some(tx => !tx.rates?.[localCurrency]);
+    const totalFiatAmountPerDay = sumTransactionsFiat(
+        transactions,
+        localCurrency,
+        historicFiatRates,
+    );
+    const isMissingFiatRates = transactions.some(tx => {
+        const fiatRateKey = getFiatRateKey(
+            tx.symbol,
+            localCurrency as FiatCurrencyCode,
```

Comment:
I think this cast is unnecessary, This component for some reason accepts localCurrency as string but the data are sourced from state which is already typed as `FiatCurrencyCode`


I assume the `localCurrency` prop of this component can be changed from `string` to `FiatCurrencyCode`